Add `make doc` target to top-level Makefile

gvancuts opened the pull request:
There are multiple commits in this PR as there were some (unrelated) problems that I had to fix in order to test my changes.

**Note:** that this does **not** enable the documentation build by default but with this, we're one very short step away from enabling it (and which would allow us to close #218).

**Note2:** the documentation build is still broken since we merged #387 . A fix for that is being reviewed on the mailing list so it's not being addressed in this PR.

NanlinXie commented:
Can one of the admins verify this patch?

dbkinder commented:
And of course, it's better not to mix different changes like this into one PR.  I'd like to immediately accept the link fix in the acrn-probe doc, but not the makefile change :(

gvancuts commented on Makefile:
```diff
@@ -36,9 +37,13 @@ misc: tools
 	mkdir -p $(MISC_OUT)
 	make -C $(T)/misc OUT_DIR=$(MISC_OUT)
 
+doc:
```
> What's the reason to change where the documentation build output goes?

So that it's consistent with the rest of the build, all build artefacts are put under the build dir. (except I could get `doxygen` to put its stuff in there so it's still done inside the `doc` folder.

gvancuts replied:
> The doc/Makefile won't notice your change here because it sets:
> `BUILDDIR      = _build`
> To complete this change, you'd need to tweak the doc/Makefile to use
> `BUILDDIR      ?= _build`

That was not needed on my system. With this patch, calling `make doc` from the top-level folder puts the final documentation inside `build`. Using `O=xxx` works too. Have you tried this on your side?

gvancuts commented:
I've updated my PR with the following modifications:
* Updated the `doc/Makefile` to use `?=` to set the `BUILDDIR`
* Remove the fix for the `acrnprobe` doc that's now in PR #410 
Oh, and rebased to `master`

gvancuts commented:
I'm also tempted to **really** add `make doc` part of the default targets so we can fix and close #218. (and avoid that we break the doc build in the future). Thoughts?

dbkinder commented:
I'd love to have doc building be part of the CI test, so +1 from me for making it part of the default targets.

gvancuts commented:
> I'd love to have doc building be part of the CI test, so +1 from me for making it part of the default targets.

I've thought about this a little more and I'm most definitely of the opinion that the CI system should build our documentation as part of the standard verification procedure. I don't think we want to force **all** users to build the documentation by default though (which would happen if we make it part of the `all` target).

With that in mind, I'd rather add a simple test to the CI system that *just* calls `make doc` (in addition to all the testing it's already doing). @NanlinXie @lijinxia , can you take care of that?

TianhuaSun commented:
Verified+1 
please install doxygen

dbkinder commented:
@NanlinXie @gvancuts Looks like the doc building dependencies aren't install in the the CI build environment:

```
make -C /workspace/acrn-hypervisor-premerge-build-nuc/acrn-hypervisor/doc html BUILDDIR=/workspace/acrn-hypervisor-premerge-build-nuc/acrn-hypervisor/build/doc
make[1]: Entering directory '/workspace/acrn-hypervisor-premerge-build-nuc/acrn-hypervisor/doc'
make[1]: [Makefile:52: html] Error 2 (ignored)

New errors/warnings found, please fix them:
==============================================

Warning: sphinx_rtd_theme missing. Use pip to install it.

Extension error:
Could not import extension breathe (exception: No module named 'breathe')
make[1]: Leaving directory '/workspace/acrn-hypervisor-premerge-build-nuc/acrn-hypervisor/doc'
```

Doc build dependencies are documented in https://projectacrn.github.io/latest/howtos/process/docbuild.html#installing-the-documentation-tools

Are the CI environment configuration files checked into GitHub somewhere? (They should)  How do we update the CI environment?
